set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java

```diff
@@ -260,6 +267,13 @@ public void validateWebUserSet(UserSet webUserSet) throws RequestBodyValidationE
 	 */
 	public void deleteUserSet(String userSetId) throws UserSetNotFoundException {
 
+		// in case it is a closed set, remove the items that are members of the Set.
+		UserSet userSet = getUserSetById(userSetId);
+		if (isOpenSet(userSet)) {
```

Comment:
this code is note doing anything, please remove it and double check if the reading of the UserSet from database is needed at this stage

Reply:
method refactored according to specifications